sklearn/utils/fixes.py

```diff
@@ -170,3 +172,36 @@ def _mode(a, axis=0):
     if sp_version >= parse_version("1.9.0"):
         return scipy.stats.mode(a, axis=axis, keepdims=True)
     return scipy.stats.mode(a, axis=axis)
+
+
+###############################################################################
+# Backport of Python 3.9's importlib.resources
+# TODO: Remove when Python 3.9 is the minimum supported version
+
+
+def _open_text(data_module, data_file_name):
+    if sys.version_info >= (3, 9):
+        return resources.files(data_module).joinpath(data_file_name).open("r")
+    else:
+        return resources.open_text(data_module, data_file_name)
+
+
+def _open_binary(data_module, data_file_name):
+    if sys.version_info >= (3, 9):
+        return resources.files(data_module).joinpath(data_file_name).open("rb")
+    else:
+        return resources.open_binary(data_module, data_file_name)
+
+
+def _read_text(descr_module, descr_file_name):
+    if sys.version_info >= (3, 9):
+        return resources.files(descr_module).joinpath(descr_file_name).read_text()
+    else:
+        return resources.read_text(descr_module, descr_file_name)
+
+
+def _path(data_module, data_file_name):
+    if sys.version_info >= (3, 9):
+        return resources.as_file(resources.files(data_module).joinpath(data_file_name))
+    else:
+        return resources.path(data_module, data_file_name)
```

Comment:
Would it make sense to commute functions' definitions with the test based for `sys.version_info >= (3, 9)`? This would be:

```python
if sys.version_info >= (3, 9):
   # Use the newest preferred interfaces chaining of `importlib`.
   _open_text =   # ... 
   _open_binary = # ...
   _read_text =   # ...
   _path =        # ...
else:
   _open_text = resources.read_text
   _open_binary = resources.open_binary
   _read_text = resources.read_text
   _path = resources.path
```

But that's probably harder to define.

Reply:
Since the arguments are not dispatched in the same way, I don't think that we will end-up with more readable knowing that we are going to drop it soonish.